src/gurt/misc.c

```diff
@@ -35,31 +64,74 @@ d_rand()
 void
 d_free(void *ptr)
 {
+	if (unlikely(track_arg != NULL)) {
+		size_t size = malloc_usable_size(ptr);
+
+		d_free_track_cb(track_arg, size);
+	}
+
 	free(ptr);
 }
 
 void *
 d_calloc(size_t count, size_t eltsize)
 {
-	return calloc(count, eltsize);
+	void *ptr;
+
+	ptr = calloc(count, eltsize);
+	if (unlikely(track_arg != NULL)) {
+		if (ptr != NULL)
+			d_alloc_track_cb(track_arg, malloc_usable_size(ptr));
```

Comment:
What if d_alloc_track_cb() isn't registered?

Reply:
seems it assumes the track_arg is registered after track_cb, but agreed that seems better to not register track_arg separately if possible.

Reply:
sure, it makes sense.